A14-channelz.md

```diff
@@ -0,0 +1,1051 @@
+NOT of its descendent channels.  This is to say that an ancestor channel
+is not an aggregator for descendent channels.  The target of a channel or
+subchannel may be different from that of its parent, or not present at all.  The
+channel state may be different from the ancestor channel state.   The number of
+calls started, succeeded, and failed should reflect if calls were specifically
+tied to the channel.
+
+The number of calls started, succeeded, and failed on a channel gives insight
+into the activity of the channel.  Subtracting the number of failed and
+succeeded calls from the number started indicates how many are in flight.  The
+total number of started calls indicates how active the channel has been over
+time.  The ratio between succeeded calls and failed calls gives a rough
+estimation of how healthy the channel has been.  The last call started timestamp
+indicates how close the channel is towards entering the idle state.
+
+#### Channel Trace
+
+Channels and Subchannels also have a concept of a channel trace.  This includes
+interesting events that have happened on the channel recently.  Example events
+include entering different channel states, name resolution, load balancing
+changes, unreachability, etc.  As mentioned earlier, this tracing info is
+specific to the channel, and not an aggregation of descendent channel traces.
+
+The channel traces may include ids pointing to channels and subchannels outside
+of the current DAG.  Since channel traces use the same ids as the channels 
+themselves, this allows a user to query for more information about a channel 
+given its trace.  For example, the trace may mention that name resolution 
+failed on a different channel outside of the tree.
+
+### Subchannels
+
+To keep the size of the Channel message down, only the ChannelRefs and 
+SubchannelRefs will be included in the channel or subchannel.  These refs can
+be used to walk the hierarchy.  Subchannels are not included in the top level
+channels, and thus do not need to be paginated.  A user may query for all the
+subchannels in parallel. As mentioned above, if there are any Sockets on the 
+channel, there will be no descendent channels.   This invariant exists to 
+simplify the tree structure, and more closely match the known implementations of
+gRPC.
+
+The channel DAG represents a "has-a" rather than a "is-a" relationship.  In the
+common case, there will be many subchannels.  These subchannels will be used to
+balance RPCs for a given target.
+
+Alternatively, a given channel's load balancer may itself make RPCs to gather
+data about which subchannels to prefer.  Thus, the load balancing subchannels
+are owned by a parent, but do not share its target.
+
+![channel hierarchy 2][channel hierarchy 2]
+
+Lastly, subchannels may not be shared across different channel trees.  Each
+channel has at most one parent.  If a channel tree is used between multiple
+other channels, it must reside in its own tree.  For example, a name resolver
```

Comment:
What is a "name resolver channel"?

Reply:
A channel that resolves names, likely using gRPC as an underlying transport.  The NR is just an example, it can be anything (such as load balancing, or security, or etc.)